### ISSUE TITLE
SC2034 does not respect allexport

### ISSUE THREAD

**Issue by nicerobot** [koalaman/shellcheck#1986]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or "online"): 0.7.1
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2034)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
set -o allexport
I_AM_EXPORTED=42

```

#### Here's what shellcheck currently says:

```sh
$ shellcheck myscript
 
Line 3:
I_AM_EXPORTED=42
^-- SC2034: I_AM_EXPORTED appears unused. Verify use (or export if used externally).
```

#### Here's what I wanted or expected to see:

```shell
$ shellcheck myscript
No issues detected!
```
